native-sql-engine/core/src/main/scala/org/apache/spark/sql/execution/ColumnarBroadcastExchangeExec.scala

```diff
@@ -82,11 +84,23 @@ case class ColumnarBroadcastExchangeExec(mode: BroadcastMode, child: SparkPlan)
   @transient
   lazy val completionFuture: scala.concurrent.Future[broadcast.Broadcast[Any]] =
     promise.future
-
+  
```

Comment:
Already put this piece of code into shim layer. 
TODO: 
Another thought is: directly apply this piece of code to gazelle, no need to distinguish spark3.1 & 3.2.